### DISCUSSION TITLE
How to use `ZLIBNG_VER_STATUS` in code

### DISCUSSION THREAD

**Discussion by pmqs** [zlib-ng/zlib-ng#1517] [Q&A]
With the 2.1.2 release just out the value of `ZLIBNG_VER_STATUS` has morphed from a number to the letter `F`. This is documented in the definition of  `ZLIBNG_VER_STATUS` 

```
#define ZLIBNG_VER_STATUS F         /* 0=devel, 1-E=beta, F=Release */
```

Preciously I  assumed it was a integer and my code does something like this

```
int value = ZLIBNG_VER_STATUS;
```

To deal with it now being either a number or a letter,  I need to read the value as a string, but C macro magic doesn't seem to give me a way to wrap the value of `ZLIBNG_VER_STATUS` in double quotes. 

Any ideas?

**Comment by Dead2** [comment]
Would something like this work for you?

```C
#include <stdio.h>

#define STRINGIFY(x) STRINGIFY_HELPER(x)
#define STRINGIFY_HELPER(x) #x

#define ZLIBNG_VER_STATUS F

int main() {
    const char* verStatusString = STRINGIFY(ZLIBNG_VER_STATUS);

    printf("Macro literal value: %s\n", verStatusString);

    return 0;
}
```

**Comment by pmqs** [comment]
Hey @Dead2 already been down that path but it is not quite what I need 

The use-case here is where these macros get processed by C to Perl interface code that auto-generates Perl variables that match C symbols. To make that happen without jumping through hoops I need to end up with the equivalent of `#define ZLIBNG_VER_STATUS "F"` in my interface code

Last thing tried was this, but that doesn't work.

```
#include <stdio.h>

#define STRINGIFY(x) STRINGIFY_HELPER(x)
#define STRINGIFY_HELPER(x) #x

#define ZLIBNG_VER_STATUS F

#define keep ZLIBNG_VER_STATUS
#undef ZLIBNG_VER_STATUS
#define ZLIBNG_VER_STATUS STRINGIFY(ZLIBNG_VER_STATUS);

int main() {
    const char* verStatusString = ZLIBNG_VER_STATUS;

    printf("Macro literal value: %s\n", verStatusString);

    return 0;
}
```

that ends up with

```
Macro literal value: ZLIBNG_VER_STATUS
```

If this cannot be achieved with the C pre-processor, I can do a workaround.

**Comment by pmqs** [comment]
I've worked around the issue. Closing.
